File: sentimental-mario-less/mario.py
import cs50


def main():
    while (True):
        height = cs50.get_int("Height: ")
        if (height > 0 and height < 9):
            break
    print_right_pyramid(height)


def print_right_pyramid(height):
    for i in range(0, height):
        for j in range(1, height + 1):
            if (j < height - i):
                print(" ", end="")
            else:
                print("#", end="")
        print()


if __name__ == "__main__":
    main()

File: mario-less/mario.c
#include <cs50.h>
#include <stdio.h>

void print_right_pyramid(int height);
void print_left_pyramid(int height);

int main(void)
{
    const int MAX_HEIGHT = 8;
    int height;
    do
    {
        height = get_int("Height: ");
    }
    while (height <= 0 || height > MAX_HEIGHT);
    print_right_pyramid(height);
}

void print_right_pyramid(int height)
{
    for (int i = 0; i < height; i++)
    {
        for (int j = 1; j <= height; j++)
        {
            if (j < height - i)
            {
                printf(" ");
            }
            else
            {
                printf("#");
            }
        }
        printf("\n");
    }
}

void print_left_pyramid(int height)
{
    for (int i = 0; i < height; i++)
    {
        for (int j = height - i; j <= height; j++)
        {
            printf("#");
        }
        printf("\n");
    }
}
